Explorer/Assets/Scripts/SceneRuntime/Factory/SceneRuntimeFactory.cs

```diff
@@ -24,6 +25,26 @@ public enum InstantiationBehavior
         private readonly JsCodeResolver codeContentResolver;
         private readonly Dictionary<string, string> sourceCodeCache;
 
+        private static readonly IReadOnlyCollection<string> JS_MODULE_NAMES;
+        private readonly IJsSceneSourceCode jsSceneSourceCode =
```

Comment:
It looks like the purpose for this is debugging but it is injected directly into the runtime class with a bit of obscured logic (returning `null` means using `production` code). Can we isolate it and extract from the normal flow?

Reply:
WDYM about returning null and the isolation? This code works only in Editor